File: Java/Easy/2696.java
class Solution {
    public int minLength(String s) {
        // greedy approach using a stack
        Stack<Character> stack = new Stack<>();

        for (char c : s.toCharArray()) {
            if (stack.size() >= 1) {
                if ((stack.peek() == 'C' && c == 'D') || (stack.peek() == 'A' && c == 'B')) {
                    stack.pop();
                } else {
                    stack.push(c);
                }
            } else {
                stack.push(c);
            }
        }

        return stack.size();
    }
}

File: Python/Easy/2696.py
class Solution:
    def minLength(self, s: str) -> int:
        # use a greedy approach with a stack
        stack = []

        for c in s:
            if ("C" in stack) or ("A" in stack):
                if (c == "D" and stack[-1] == "C") or (c == "B" and stack[-1] == "A"):
                    stack.pop()
                else:
                    stack.append(c)
            else:
                stack.append(c)

        return len(stack)
